### PR TITLE
fix: Revamp ZRC-1 NFT Standard to comply with ERC-777 interface

### PR DESCRIPTION
**Pull request by evesnow91**

### CODE REVIEW

File: reference/nonfungible-token.scilla [hunk 1]
```diff
@@ -64,257 +71,472 @@ let makeErrorEvent =
...
+  msgs = one_msg msg_to_sender;
+  send msgs
+end
+
+(* Interface transitions *)
+
+(* @dev:    Add or remove approved minters. Only contract_owner can approve minters. *)
+(* @param:  minter      - Address of the minter to be approved or removed            *)
+(* Returns error event CodeNotAuthorised if _sender is not contract_owner.           *)
+transition configureMinter(minter: ByStr20)
```
Comment (edisonljh) [inline reference/nonfungible-token.scilla]: this transition looks very confusing to me. 
It's very error-prone. Is this from an existing contract standard? I can't find it on github https://github.com/0xjac/ERC777/blob/devel/contracts/ERC20Token.sol

File: reference/nonfungible-token.scilla [hunk 2]
```diff
@@ -64,257 +71,472 @@ let makeErrorEvent =
...
+      owned_token_count[to] := new_count;
+      (* Add token_uri for token_id *)
+      token_uris[token_id] := token_uri;
+      (* Add to total_supply *)
+      current_supply <- total_supply;
+      new_supply = builtin add current_supply one;
+      total_supply := new_supply; 
       (* Emit success event *)
-      e = {_eventname: "MintSuccess"; by: _sender; recipient: to; token: tokenId};
+      e = {_eventname: "MintSuccess"; by: _sender; recipient: to;
```
Comment (edisonljh) [inline reference/nonfungible-token.scilla]: I think we have to standardise on the case for `_eventname`. Some of them are camelCase but some are PascalCase

File: reference/nonfungible-token.scilla [hunk 3]
```diff
@@ -64,257 +71,472 @@ let makeErrorEvent =
...
+        e = {_eventname: "AddApprovalSuccess"; initiator: _sender; approved_spender: to; token_id: token_id};
+        event e;
+        msg_to_sender = { _tag : "addApprovalSuccessCallBack"; _recipient : _sender; _amount : Uint128 0; 
+                          approved_spender : to; token_id : token_id };
+        msgs = one_msg msg_to_sender;
+        send msgs
+      | Some addr =>
+        delete token_approvals[token_id];
+        (* Emit event *)
+        e = {_eventname: "RemoveApprovalSuccess"; initiator: _sender; removed_spender: to; token_id: token_id};
```
Comment (edisonljh) [inline reference/nonfungible-token.scilla]: Incosistent case

File: reference/nonfungible-token.scilla (hunk 1 above)
Comment (evesnow91) [reply reference/nonfungible-token.scilla]: I needed a way to approve and remove `minters` role from the contract, so I've added this `configureMinter` transition.
Currently, `mint` and `burn` are both privileged transitions that can only be called by `minters`.
Any suggestion to make this less error-prone?

File: reference/nonfungible-token.scilla (hunk 2 above)
Comment (evesnow91) [reply reference/nonfungible-token.scilla]: I believe that all `_eventname` are currently in PascalCase.

**Pull request merged**